src/Component/src/Metadata/Extractor/PhpFileResourceExtractor.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Extractor;
+
+use Sylius\Resource\Metadata\ResourceMetadata;
+
+final class PhpFileResourceExtractor extends AbstractResourceExtractor
+{
+    /**
+     * @inheritdoc
+     */
+    protected function extractPath(string $path): void
+    {
+        $resource = $this->getPHPFileClosure($path)();
+
+        if (!$resource instanceof ResourceMetadata) {
+            return;
+        }
+
+        $resourceReflection = new \ReflectionClass($resource);
+
+        foreach ($resourceReflection->getProperties() as $property) {
+            $property->setAccessible(true);
+            $resolvedValue = $this->resolve($property->getValue($resource));
+            $property->setValue($resource, $resolvedValue);
+        }
+
+        $this->resources = [$resource];
```

Comment:
Function `public function getResources(): array` is meant to extract multiple resources by multiple paths, but this line overrides `$resources`.

Is this planned behavior?

Reply:
we configure a directory on the config, but the Symfony Finder will extract files.

Reply:
Have a look here
https://github.com/api-platform/core/pull/7017/files#diff-9623e12b027ed7457a5b161569c35b27631c53b0a510e6f6e615ade3bb0353c5R427

Reply:
```patch
        $this->resources = [];
        foreach ($this->paths as $path) {
---         $this->extractFromPath($path);
       }
        $this->resources = [];
        foreach ($this->paths as $path) {
+++        $this->resources[] = $this->extractFromPath($path);
        }